pkg/build/generator/generator.go

```diff
@@ -416,6 +416,20 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 	build.Labels[buildapi.BuildConfigLabelDeprecated] = bcCopy.Name
 	build.Labels[buildapi.BuildConfigLabel] = bcCopy.Name
 
+	// Set the queued label for builds created from "serial" and "rapid" build configuration.
+	// The "serial" scheduling policy is default policy if not specified.
+	policy := bc.Spec.SchedulePolicy
+	if policy == nil {
+		build.Labels[buildapi.BuildQueuedLabel] = "true"
+	} else {
+		switch *policy {
+		case buildapi.BuildSchedulePolicySerial:
+			build.Labels[buildapi.BuildQueuedLabel] = "true"
+		case buildapi.BuildSchedulePolicySerialLatestOnly:
+			build.Labels[buildapi.BuildQueuedLabel] = "false"
```

Comment:
debatable whether this should have the queued label..  certainly when you have just 2 builds (one running, one next) it makes sense to call that 2nd build "queued".

when you have 3, so the middle one will never run, calling it queued is more debatable.